addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV6SessionManager.java

```diff
@@ -331,8 +351,10 @@ public void keep_alive(int periodic_interval_ms) {
     /**
      * The main state machine of the session handshake. This will start a timer
      * that will reset the handshake if we do not get a satisfying response in time.
+     *
+     * @throws InterruptedException
      */
-    private void session_handshake_process() {
+    private void session_handshake_process() throws InterruptedException {
```

Comment:
The alternative is to ignore the "Thread.sleep" InterruptedException. In this new implementation I pass this Exception all the way up the the corresponding Thread runnables, where the exception basically cause the end of the thread (which is what the user requested) in a controlled way.